File: client/src/types/member.ts
export type ServiceCategory = {
  id: number;
  key: string;
  name: string;
};

export type MemberSkill = {
  id: number;
  memberId: string;
  serviceCategoryId: number;
  note?: string;
  serviceCategory: ServiceCategory;
};

export type MemberNeed = {
  id: number;
  memberId: string;
  serviceCategoryId: number;
  note?: string;
  serviceCategory: ServiceCategory;
};

export type MemberAvailabilitySlot = {
  id: number;
  memberId: string;
  dayOfWeek: number;
  startHour: number;
  endHour: number;
  mode: number;
};

export type Photo = {
  id: number;
  url: string;
  publicId?: string;
  memberId: string;
};

export type Member = {
  id: string;
  displayName: string;
  about?: string;
  avatarUrl?: string;
  city?: string;
  countryCode?: string;
  isProfilePublic: boolean;
  createdAtUtc: string;
  lastActiveAtUtc: string;
  photos: Photo[];
  offeredSkills: MemberSkill[];
  needsHelpWith: MemberNeed[];
  availabilitySlots: MemberAvailabilitySlot[];
};

File: API/Entities/Member.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace API.Entities;

public class Member
{
    public string Id { get; set; } = null!;
    public required string DisplayName { get; set; }
    public string? About { get; set; }
    public string? AvatarUrl { get; set; }
    public string? City { get; set; }
    public string? CountryCode { get; set; }
    public bool IsProfilePublic { get; set; } = true;
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime LastActiveAtUtc { get; set; } = DateTime.UtcNow;

    [JsonIgnore]
    public List<Photo> Photos { get; set; } = [];
    public List<MemberSkill> OfferedSkills { get; set; } = [];
    public List<MemberNeed> NeedsHelpWith { get; set; } = [];
    public List<MemberAvailabilitySlot> AvailabilitySlots { get; set; } = [];

    [JsonIgnore]
    [ForeignKey(nameof(Id))]
    public AppUser User { get; set; } = null!;
}
